### ISSUE TITLE
macOS Catalina 10.15.5 reports hale studio.app 4.0.0 as damaged

### ISSUE THREAD

**Issue by fjlopez** [halestudio/hale#847]
Expected behaviour
----
Run **hale studio.app** obtained from the 4.0.0 release **[.dmg](https://github.com/halestudio/hale/releases/download/4.0.0/hale-studio-4.0.0-macosx.cocoa.x86_64.dmg)** in **macOS Catalina 10.15.5**.

Current Behaviour 
----
**macOS Catalina 10.15.5** reports **hale studio.app**  as damaged, and, hence, can't be opened, _even with Gatekeeper disabled_.

<img width="488" alt="Screenshot 2020-07-06 at 12 39 07" src="https://user-images.githubusercontent.com/3294927/86584763-b6674c00-bf85-11ea-83ea-88d158d8e938.png">

Steps to reproduce 
----

1. Open a terminal
1. Execute `sudo spctl --master-disable` to disable Gatekeeper and then allow run apps downloaded anywhere.
1. Download `hale studio 4.0.0.dmg` from https://www.wetransform.to/downloads/
1. Open the dmg file. 
1. Move `hale studio.app` to the `Applications` folder. 
1. Go to the `Applications` folder
1. Open `hale studio.app`

**Comment by florianesser** [comment]
Hi @fjlopez, thanks for using hale studio and reaching out to us!

hale studio 4.0.0 still needs Java 8 to run. The macOS version doesn't come with a packaged JDK and a typical problem is that JDK 8 is not available. Do you have JDK 8 installed? If you open a console, what does the command `java -version` return?

**Comment by fjlopez** [comment]
I use `jenv` to manage different JDK versions. I have tested it with the Oracle JDK 8 as follows:

```bash
$ jenv global oracle64-1.8.0.144
$ javac -version 
javac 1.8.0_144
$ java -version
java version "1.8.0_144"
Java(TM) SE Runtime Environment (build 1.8.0_144-b01)
Java HotSpot(TM) 64-Bit Server VM (build 25.144-b01, mixed mode)
$ echo $JAVA_HOME
/Users/javier/.jenv/versions/oracle64-1.8.0.144
$ sudo spctl --master-disable
$ open -a "Hale Studio"
```
and then fails

**Comment by florianesser** [comment]
Thanks for the info, we'll look into it.

**Comment by florianesser** [comment]
While investigating this problem, I found that hale studio 4.0.0 doesn't start up on macOS even when the error reported in this issue does not occur (see #849). I think it makes sense to work on the start problem first and see if that helps to resolve this issue.

**Comment by JohannaOtt** [comment]
@fjlopez A potential workaround is described [here](https://github.com/halestudio/hale/issues/849#issuecomment-899680614)

**Comment by thorsten-reitz** [comment]
@fjlopez The new release (4.1.0) has been tested thoroughly with current MacOS versions and models. Please let us know if you can't work with that. In that case we will re-open this ticket.

**Comment by fjlopez** [comment]
Don't work yet. My macOS version is 11.6.1 Big Sur. 

<img width="270" alt="Screenshot 2021-12-05 at 12 54 50" src="https://user-images.githubusercontent.com/3294927/144745472-7f2d1db3-0fa5-4022-be43-23631373dc3e.png">

This version is not correctly signed and notarized, the output of this command is:

```sh
$  spctl -a -v hale\ studio.app               
hale studio.app: code has no resources but signature indicates they must be present
```

when something like:

```sh
$  spctl -a -v hale\ studio.app               
hale studio.app: accepted
```

was expected ([source](https://help.apple.com/xcode/mac/current/#/dev1cc22a95c)).

**Comment by kapil-agnihotri** [comment]
@fjlopez, thank you for your feedback. 
Please follow the `installation instructions` for Mac as document in the [release 4.1.0 page](https://github.com/halestudio/hale/releases/tag/4.1.0).

Steps to run hale studio in Mac
If you are using Mac with intel chip, then to be able to run the application after the installation, you must execute the below command in the terminal.
```
 xattr -cr '/Applications/hale studio.app'
```
Then launch hale studio from `Applications`

In case you are using Mac with M1 chip then let us know.

**Comment by fjlopez** [comment]
That is an unsafe recommendation as`xattr -cr` is used in zero days attacks such as [xcsset](https://www.trendmicro.com/en_us/research/20/h/xcsset-mac-malware--infects-xcode-projects--uses-0-days.html). 

Please, review your notarisation process (e.g. [here is an answer that address the case "code has no resources but signature indicates they must be present"](https://blog.adoptopenjdk.net/2020/05/a-simple-guide-to-notarizing-your-java-application/)).

**Comment by thorsten-reitz** [comment]
@kapil-agnihotri Do you think this issue can be closed with the 5.x releases?

**Comment by JohannaOtt** [comment]
@stempler Do you think this issue can be closed with the 5.x releases?

**Comment by stempler** [comment]
No, there is no notarization process set up. So far a blocker was that the required tools require macOS to run, which was not available in our CI infrastructure. Now that GitHub Actions is used there is the possibility to [also run on macOS](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories). [Here](https://federicoterzi.com/blog/automatic-code-signing-and-notarization-for-macos-apps-using-github-actions/) is also a guide on how to do the notarization using GitHub Actions that may be applicable for us.
